scripts/pelorus-operator-patches/07_spec_description.diff

```diff
@@ -0,0 +1,89 @@
++++ config/crd/bases/charts.pelorus.konveyor.io_pelorus.yaml	2022-12-16 11:40:22.725983634 +0100
+@@ -15,29 +15,76 @@
+   - name: v1alpha1
+     schema:
+       openAPIV3Schema:
+-        description: Pelorus is the Schema for the pelorus API
++        description: >-
++          Configure a running instance of Pelorus
++        type: object
+         properties:
+           apiVersion:
+-            description: 'APIVersion defines the versioned schema of this representation
+-              of an object. Servers should convert recognized schemas to the latest
+-              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
++            description: >-
++              APIVersion defines the versioned schema of this representation
++              of an object. Servers should convert recognized schemas to the
++              latest internal value, and may reject unrecognized values. More
++              info:
++              https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+             type: string
+           kind:
+-            description: 'Kind is a string value representing the REST resource this
+-              object represents. Servers may infer this from the endpoint the client
+-              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
++            description: >-
++              Kind is a string value representing the REST resource this
++
++              object represents. Servers may infer this from the endpoint the
++              client
++
++              submits requests to. Cannot be updated. In CamelCase. More info:
++              https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+             type: string
+           metadata:
+             type: object
+           spec:
+-            description: Spec defines the desired state of Pelorus
++            description: >-
++              PelorusSpec defines state of the Pelorus application and allows
++              to configure for the desired workflow. More information about
++              Pelorus configuration that can be used in the Spec section is
++              available at:
++              https://pelorus.readthedocs.io/en/latest/Configuration/
+             type: object
+-            x-kubernetes-preserve-unknown-fields: true
++            properties:
++              exporters:
++                description: >-
++                  References configuration for the Pelorus exporters. More
++                  info about exporters configuration:
++                  https://pelorus.readthedocs.io/en/latest/Configuration/#configuring-exporters-details
++
++                    Example:
```

Comment:
here I would remove the tab from "examples:" (consequently, remove one from each of the following lines)

Reply:
Instead of the description, could it be a object in another **properties** section?

Reply:
That can't be done. We have problem when we want to use properties section in here. The OCP evaluates every entry in properties and it's not matching openAPIV3Schema because it's Helm values.yaml which is in fact json.

I have tried with:
```
              properties:
                exporters:
                  description: >-
                    References configuration for the Pelorus exporters. More
                    info about exporters configuration:
                    https://pelorus.readthedocs.io/en/latest/Configuration/#configuring-exporters-details
                  type: object
                  properties:
                    instances:
                      description: >-
                        Exporter instances configuration. More information:
                        https://pelorus.readthedocs.io/en/latest/Configuration/
                      type: array
                      items:
                        type: object
                      properties:
                        app_name:
                          description: 'Exporter name, must start with "- ".'
                          type: string
                        enabled:
                          description: 'If set to false, exporter won''t be provisioned.'
                          type: boolean
                        env_from_configmaps:
                          description: >-
                            ConfigMap name(s) with configuration used by the
                            exporter instance.
                          type: object
                        env_from_secrets:
                          description: >-
                            Secret name(s) with secrets used by the exporter
                            instance.
                          type: object
                        exporter_type:
                          description: >-
                            Exporter type as in the:
                            https://pelorus.readthedocs.io/en/latest/Configuration/#configuring-exporters-overview
                          type: string
```

But OCP was creating malformed values, because there is no such type that can fit our list (problematic - at the beginning of the line):
```
- app_name: first
- app_name: second
```

Reply:
:( alright, I think the way you did is fine, and in the docs it is well formatted as well

Reply:
We can always find better solution in separate PR

Reply:
Resolving as it's now tracked as an issue